packages/contracts/contracts/generator/GenArt721GeneratorV0.sol

```diff
@@ -0,0 +1,331 @@
+     * @return The HTMLRequest for the token.
+     */
+    function getTokenHtmlRequest(
+        address coreContractAddress,
+        uint256 tokenId
+    ) internal view returns (HTMLRequest memory) {
+        _onlySupportedCoreContract(coreContractAddress);
+
+        uint256 projectId = tokenId / ONE_MILLION;
+        // This will revert for older contracts that do not have an override set.
+        bytes32 dependencyNameAndVersion = dependencyRegistry
+            .getDependencyNameAndVersionForProject(
+                coreContractAddress,
+                projectId
+            )
+            .stringToBytes32();
+
+        bytes32 tokenHash;
+        if (tokenHash == bytes32(0)) {
+            try
+                IGenArt721CoreTokenHashProviderV1(coreContractAddress)
+                    .tokenIdToHash(tokenId)
+            returns (bytes32 _tokenHash) {
+                tokenHash = _tokenHash;
+            } catch {
+                // Noop try again for older contracts.
+            }
+        }
+
+        if (tokenHash == bytes32(0)) {
+            try
+                IGenArt721CoreTokenHashProviderV0(coreContractAddress)
+                    .showTokenHashes(tokenId)
+            returns (bytes32[] memory tokenHashes) {
+                tokenHash = tokenHashes[0];
+            } catch {
+                revert("Unable to retrieve token hash.");
+            }
+        }
+
+        HTMLTag[] memory headTags = new HTMLTag[](2);
+        headTags[0].tagOpen = "<style>";
+        headTags[0]
+            .tagContent = "html{height:100%}body{min-height:100%;margin:0;padding:0}canvas{padding:0;margin:auto;display:block;position:absolute;top:0;bottom:0;left:0;right:0}";
+        headTags[0].tagClose = "</style>";
+
+        headTags[1].tagContent = abi.encodePacked(
+            'let tokenData = {"tokenId":"',
+            Strings.toString(tokenId),
+            '"',
+            ',"hash":"',
+            Strings.toHexString(uint256(tokenHash)),
+            '"}'
+        );
+        headTags[1].tagType = HTMLTagType.script;
+
+        HTMLTag[] memory bodyTags = new HTMLTag[](3);
+        // Get script count and preferred CDN for the dependency.
+        (
+            ,
+            ,
+            string memory preferredCDN,
+            ,
+            ,
+            ,
+            ,
+            ,
+            uint24 scriptCount
+        ) = dependencyRegistry.getDependencyDetails(dependencyNameAndVersion);
+
+        // If no scripts on-chain, load the script from the preferred CDN.
+        if (scriptCount == 0) {
+            bodyTags[0].tagOpen = abi.encodePacked(
+                '<script type="text/javascript" src="',
+                preferredCDN,
+                '">'
+            );
+            bodyTags[0].tagContent = "// Noop"; // ScriptyBuilder requires scriptContent for this to work
+            bodyTags[0].tagClose = "</script>";
+        } else {
+            bytes memory dependencyScript = this.getDependencyScript(
```

Comment:
any reason to not make `getDependencyScript` public and remove `this.` here? should be cheaper to make call internal.

Reply:
Mainly I thought it might be nice to be able to get the full dependency script on its own easily but I guess we're returning bytes so it wouldn't be immediately human readable anyway. Will go ahead and make internal and we can always add another public view function in the future if we want.

Reply:
I'm actually gonna rename these to getDependencyScriptBytes and getProjectScriptBytes and then add external views to easily grab the full scripts without concatenating on-chain.

Reply:
cool - yeah and just to double check, was suggesting to change from `external` to `public`, so we have both external function + ability to call as internal as well.

Reply:
Ah that makes sense! I was sort of conflating external and public so I thought you were suggesting to make it internal. I think I may keep the update anyway so we have a function where input and output are both human readable.